ci-operator/step-registry/cucushift/chainupgrade/toimage/cucushift-chainupgrade-toimage-commands.sh

```diff
@@ -282,9 +282,13 @@ function rhel_upgrade(){
 # Extract oc binary which is supposed to be identical with target release
 function extract_oc(){
     echo -e "Extracting oc\n"
-    local retry=5 tmp_oc="/tmp/client-2"
+    local rhel_version retry=5 tmp_oc="/tmp/client-2" binary='oc'
     mkdir -p ${tmp_oc}
-    while ! (env "NO_PROXY=*" "no_proxy=*" oc adm release extract -a "${CLUSTER_PROFILE_DIR}/pull-secret" --command=oc --to=${tmp_oc} ${TARGET});
+    rhel_version=$(uname -r | sed -nE 's/.*el([0-9]*).*/\1/p')
+    if (( TARGET_MINOR_VERSION > 15 )) && (( rhel_version > 8 )) ; then
```

Comment:
I think here should be `(( rhel_version == 8 ))`

Reply:
no, on rhel 8 we need do nothing, only on rhel 9 we need to extract oc.rhel8

Reply:
per my local testing, when extracting `oc` from a 4.16 payload on a rhel8 host, it will extract a oc which only works on rhel9, so need to specify `oc.rhel8`. Are you sure your statement is right?

Reply:
I just go through the [slack](https://redhat-internal.slack.com/archives/CF8SMALS1/p1714967731069929) thread again, you are right.